make-dist

```diff
@@ -1,7 +1,17 @@
 #!/bin/bash -e
 
+SCRIPTNAME="$(basename "${0}")"
+BASEDIR="$(readlink -f "$(dirname "${0}")")"
+
 if [ ! -d .git ]; then
-    echo "no .git present.  run this from the base dir of the git checkout."
+    echo "$SCRIPTNAME: Full path to the script: $BASEDIR/$SCRIPTNAME"
+    echo "$SCRIPTNAME: No .git present. Run this from the base dir of the git checkout."
+    exit 1
+fi
+
+if [[ "$BASEDIR" == *:* ]] ; then
```

Comment:
This test is not quire right, I believe. This is doing a string compare to ":". It doesn't do a subset check.

It's also not clear why a colon in the script's path would break anything. POSIX/Linux does not care if a path has a colon but `/bin/sh` may care if that path appears in `$PATH`.

Reply:
@batrick Comparing a string to `*:*` is effectively a subset check and I have tested that it works, but I'm not clinging to it and can change it to a regex if you like?

The reason why the colon breaks the script is related to nodejs and is explained in the linked ticket. I believe the bug is in nodejs, but since the nodejs folks have refused to fix it, I believe it makes sense to add this sanity check here so users get a sensible error message at the beginning instead of a bizarre nodejs failure in the middle of the script run.

Reply:
> @batrick Comparing a string to `*:*` is effectively a subset check and I have tested that it works, but I'm not clinging to it and can change it to a regex if you like?

Ah! My eyes aren't that good. `*` looked like `"` to me.

> The reason why the colon breaks the script is, in my opinion, a bug in nodejs (or nodejs-related tooling): the tracker ticket contains details and a link to the upstream nodejs bug. Since the nodejs folks have refused to fix the issue, it makes sense **in my mind** to add this sanity check here so users get a sensible error message at the beginning, instead of a bizarre nodejs failure in the middle, of the script run.

very well but can you elucidate to that in a comment. A curious newcomer may be baffled by this code.

Reply:
ok; done